src/hooks/useBookmarks.js

```diff
@@ -4,25 +4,26 @@ const useBookmarks = () => {
   const [bookmarkList, setBookmarkList] = useState([]);
 
   useEffect(() => {
-    const getAllBookmark = (nodeItems) => {
+    function getAllBookmark(nodeItems) {
       const newBookmarkList = [];
+      const stack = [...nodeItems];
 
-      function recursive(node) {
+      while (stack.length) {
+        const node = stack.pop();
         if (Array.isArray(node)) {
-          node.forEach((item) => recursive(item));
+          stack.push(...node);
         } else if (typeof node === "object" && node !== null) {
           if (node.children) {
-            recursive(node.children);
+            stack.push(...node.children);
```

Comment:
```suggestion
            stack.push(...children);
```
기존의 재귀 함수 방식 대신, while 루프와 stack을 활용한 반복문 방식으로 변경하셔서 성능과 유지 보수차원에서 좋은 코드인 것 같습니다! 추가적으로 구조 분해 할당을 const { title, url, children } = node; 이렇게 적용하셔서 node.children, node.title, node.url을 children, title, url로 변경하는 방법도 가독성에 좋을 것 같아 제안드립니다. 감사합니다!

Reply:
그 생각을 해보지 못했던 것 같습니다. 구조분해할당을 사용해서 가독성을 더 올릴 수도 있었을 것 같다는 의견 감사합니다.